x/mongo/driver/topology/pool_test.go

```diff
@@ -840,6 +840,77 @@ func TestPool(t *testing.T) {
 			p1.close(context.Background())
 			p2.close(context.Background())
 		})
+		t.Run("bumps the connection idle deadline", func(t *testing.T) {
+			t.Parallel()
+
+			cleanup := make(chan struct{})
+			defer close(cleanup)
+			addr := bootstrapConnections(t, 1, func(nc net.Conn) {
+				<-cleanup
+				_ = nc.Close()
+			})
+
+			d := newdialer(&net.Dialer{})
+			p := newPool(poolConfig{
+				Address:     address.Address(addr.String()),
+				MaxIdleTime: 1 * time.Millisecond,
+			}, WithDialer(func(Dialer) Dialer { return d }))
```

Comment:
Could `defer p.close(context.Background()` here instead of calling on L875? Same with test below.

Reply:
Yeah, can definitely do that. Will update.

Reply:
Updated.